pennylane/workflow/qnode.py

```diff
@@ -763,6 +763,11 @@ def get_gradient_fn(
         """Determine the best differentiation method, interface, and device
         for a requested device, interface, and diff method.
 
+        .. warning::
+
+            This function is deprecated and will be removed in v0.42. Instead, use

```

Comment:
```suggestion
            This function is deprecated and will be removed in v0.43. Instead, use

```

Or a future version (?)